codemods/ts-morph/next/13/replace-next-head/index.ts

```diff
@@ -138,9 +160,44 @@ export const buildContainer = <T>(initialValue: T) => {
 
 type Container<T> = ReturnType<typeof buildContainer<T>>;
 
+const isDefinedOnTheTopLevel = (
+	identifier: Identifier,
+	topLevelVariablesContainer: Container<Definition[]>,
+) => {
+	//  alternative
+	//  return definition.getFirstAncestorByKind(SyntaxKind.Block) === undefined;
+	const definitions = identifier.getDefinitionNodes() as Definition[];
+	const topLevelDefinitions = topLevelVariablesContainer.get();
+	const parent = identifier.getParent();
+	const identifierChildIndex = identifier.getChildIndex();
+
+	// if identifier is within property access expression "obj.a.b", check only for its root "obj"
+	if (
+		(Node.isPropertyAccessExpression(parent) ||
+			Node.isElementAccessExpression(parent)) &&
+		identifierChildIndex !== 0
+	) {
+		return true;
+	}
+
+	// @TODO
```

Comment:
why TODO?

Reply:
i think there might be a way to infer globals from ts, currently i just check for a "process" identifier name